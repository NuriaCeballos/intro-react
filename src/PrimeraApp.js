// FC funtional components
//import React, { Fragment } from 'react'; // sin importarlo no me da error
import React from 'react';
import PropTypes from 'prop-types'; //sirve para que obligar a mandar las props


const PrimeraApp=({saludo,subtitulo})=>{

  

    return(
        <>
        <h1> { saludo } </h1>
        {/* <pre>{ JSON.stringify(saludo) }</pre> */}
        <p> { subtitulo } </p>
        </>
    );

}

PrimeraApp.propTypes={
    saludo:PropTypes.string.isRequired// con esto obligamos a 
    //mandar la porperty
}
PrimeraApp.defaultProps = {
    subtitulo: "Soy el subtitulo"
}
export default PrimeraApp;