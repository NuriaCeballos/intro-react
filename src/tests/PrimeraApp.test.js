// import '@testing-library/jest-dom/extend-expect'
// import {render} from '@testing-library/react';
import React from 'react';
import {shallow} from 'enzyme'

import PrimeraApp from '../PrimeraApp'
import '@testing-library/jest-dom'
// import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
// import {createSerializer} from 'enzyme-to-json';





describe('Pruebas en <PrimeraApp/>', () => {

    // test ( 'debe demostrar el mensaje  "Hola, Soy Nuria" ',()=>{
    //const saludo = 'Hola, soy Nuria';
    //const {getByText} = render(<PrimeraApp saludo ={saludo} />)// se destructura wrapper
    // // para sacar getByText
    //// wrapper.getByText()
     //expect( getByText(saludo )).toBeInTheDocument();
    // })

    //**** test con enzyme**** ///////*/

    test ( 'debe de mostrar <PrimeraApp /> correctament', ()=>{
        

        const saludo = 'Hola, soy Nuria'
        const wrapper = shallow(<PrimeraApp saludo = { saludo } />)

        expect (wrapper).toMatchSnapshot()
    
    })

    test ( 'debe de mostrar el suntitulo envisado por props', ()=>{

        const saludo = 'Hola, soy Nuria';
        const subTitulo = 'Soy un subtitulo';
        const wrapper = shallow(<PrimeraApp 
            saludo = {saludo }
            subtitulo = {subTitulo}
            />);

         const textoParrafo = wrapper.find('p').text()  
        //  console.log(textoParrafo);
        
        expect(textoParrafo).toBe(subTitulo)
    
    })
    
})
