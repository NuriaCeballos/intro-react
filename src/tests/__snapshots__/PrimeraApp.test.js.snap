// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> debe de mostrar <PrimeraApp /> correctament 1`] = `
<Fragment>
  <h1>
     
    Hola, soy Nuria
     
  </h1>
  <p>
    Soy el subtitulo
  </p>
</Fragment>
`;
